source/shared/core_stmt.cpp

```diff
@@ -466,7 +464,11 @@ void core_sqlsrv_bind_param( _Inout_ sqlsrv_stmt* stmt, _In_ SQLUSMALLINT param_
                     encoding == SQLSRV_ENCODING_BINARY ), "core_sqlsrv_bind_param: invalid encoding" );
 
     if( stmt->conn->ce_option.enabled && ( sql_type == SQL_UNKNOWN_TYPE || column_size == SQLSRV_UNKNOWN_SIZE )){
-        ae_get_sql_type_info( stmt, param_num, direction, param_z, encoding, sql_type, column_size, decimal_digits TSRMLS_CC );
+        // use the meta data only if the user has not specified the sql type or column size
+        sql_type = stmt->param_descriptions[param_num].sql_type;
+        column_size = stmt->param_descriptions[param_num].column_size;
+        decimal_digits = stmt->param_descriptions[param_num].decimal_digits;
+
```

Comment:
perhaps put this into a member function of param_meta_data?

Reply:
also, in the member function, check if any of this.sql_type, this.column_size, this.decimal_digits are null. Throw error if any of them is null.

Reply:
I can put these into a member function but I don't see why these fields might be null. If anything goes wrong calling SQLDescribeParam would have thrown an exception already

Reply:
Putting null checks in is to make sure the order of execution is correct (i.e., make sure SQLDescribeParam is called before core_sqlsrv_bind_param).

Reply:
I can check whether `stmt->param_descriptions[param_num]` exists. If it is, its members should be correct

Reply:
OK. Please check if stmt->param_descriptions vector size is less than param_num to check the existance of pram_descriptions[param_num]